### PR TITLE
Logging for RewardsManager and PositionManager invariants

### PR DESCRIPTION
**Pull request by ith-harvey**
# Description of change
## High level
Added logging to rewardsManager and positionManager invariants
* Added logging modifier to `rewardsPoolHandler.sol`
* Added logging modifier to `positionPoolHandler.sol`
* fixed filename path variable in `baseHandler.sol` to match readme and gitignore

### CODE REVIEW

File: tests/forge/invariants/PositionsAndRewards/handlers/RewardsPoolHandler.sol [hunk 1]
```diff
@@ -16,7 +18,7 @@ abstract contract RewardsPoolHandler is UnboundedRewardsPoolHandler, PositionPoo
         uint256 bucketIndex_,
         uint256 amountToAdd_,
         uint256 skippedTime_
-    ) external useRandomActor(actorIndex_) useRandomLenderBucket(bucketIndex_) useTimestamps skipTime(skippedTime_) {
+    ) external useRandomActor(actorIndex_) useRandomLenderBucket(bucketIndex_) useTimestamps skipTime(skippedTime_) writeRewardsLogs {
```
Comment (grandizzy) [inline tests/forge/invariants/PositionsAndRewards/handlers/RewardsPoolHandler.sol]: wouldn't make more sense to apply `writePositionLogs` modifier in addition to `writeRewardsLogs` instead calling `logPositionToFile()` inside `writeRewardsLogs`? This way we could have verbose config for rewards logs only instead always position and rewards

File: tests/README.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ Invariant test scenarios can be externally configured by customizing following e
...
-| LOGS_VERBOSITY | ERC20 ERC721 | 0 | <p> Details to log <p> 0 = No Logs <p> 1 = pool State  <p> 2 = pool State, Auctions details <p> 3 = pool State, Auctions details , Buckets details <p> 4 = pool State, Auctions details , Buckets details, Lender details <p> 5 = pool State, Auctions details , Buckets details, Lender details, Borrower details <p> Note - Log File with name `logFile.txt` will be generated in project directory|
+| LOGS_VERBOSITY | ERC20 ERC721 | 0 | <p> Details to log <p> 0 = No Logs <p> 1 = pool State  <p> 2 = pool State, Auctions details <p> 3 = pool State, Auctions details , Buckets details <p> 4 = pool State, Auctions details , Buckets details, Lender details <p> 5 = pool State, Auctions details , Buckets details, Lender details, Borrower details <p> 6 = pool State, Auctions details, Buckets details, Lender details, Borrower details, positionManager details <p> 7 = pool State, Auctions details, Buckets details, Lender details, Borrower details, positionManager details, rewardsManager details <p> Note - Log File with name `logFile.txt` will be generated in project directory|
```
Comment (prat-gpt) [inline tests/README.md]: I think pool state, auction details and borrower details are not required in position manager and rewards manager logs. So I would renaming `LOGS_VERBOSITY` to `LOGS_VERBOSITY_POOL` and create two new `LOGS_VERBOSITY_POSITION` and `LOGS_VERBOSITY_REWARD` envs for Position manager and rewards manager respectively.

File: tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol [hunk 3]
```diff
@@ -189,4 +190,61 @@ abstract contract PositionPoolHandler is UnboundedPositionPoolHandler {
...
+
+        for (uint256 i = 0; i < actors.length; i++) {
+
+            uint256[] memory tokenIds = getTokenIdsByActor(actors[i]);
+
+            if (tokenIds.length != 0) {
+                string memory actorStr = string.concat(string.concat("Actor ", Strings.toString(i)), " tokenIds: ");
+                string memory tokenIdStr;
+
+                for (uint256 k = 0; k < tokenIds.length; k++) {
```
Comment (prat-gpt) [inline tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol:220]: we can better use `printLog` [method](https://github.com/ajna-finance/contracts/blob/3052ba43fec575db901e4bb86ce61eacb03d065a/tests/forge/invariants/base/handlers/unbounded/BaseHandler.sol#L637C14-L637C22) here.

File: tests/forge/invariants/PositionsAndRewards/handlers/RewardsPoolHandler.sol (hunk 1 above)
Comment (prat-gpt) [reply tests/forge/invariants/PositionsAndRewards/handlers/RewardsPoolHandler.sol]: agree with @grandizzy here. what do u think about [these](https://github.com/ajna-finance/contracts/pull/925/files#r1255845716) verbosity envs.

File: tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol (hunk 3 above)
Comment (ith-harvey) [reply tests/forge/invariants/PositionsAndRewards/handlers/PositionPoolHandler.sol:220]: Wanted to make this all one line in the file. Forge doesn't offer any way to do this as `vm.writeLine` creates a new line every time it is called. Can change if you think code cleanliness > file output readability, but I suggest we leave it.

File: tests/forge/invariants/PositionsAndRewards/handlers/RewardsPoolHandler.sol (hunk 1 above)
Comment (ith-harvey) [reply tests/forge/invariants/PositionsAndRewards/handlers/RewardsPoolHandler.sol]: Fixed here -> https://github.com/ajna-finance/contracts/pull/925/commits/180a89ea3e909213562d4705ed6d4a35e9b91fc9

File: tests/README.md (hunk 2 above)
Comment (ith-harvey) [reply tests/README.md]: Don't have an opinion on this, could go either way @grandizzy wbu?

Happy to change to whatever you guys think is better.

File: tests/forge/invariants/base/handlers/unbounded/BaseHandler.sol [hunk 4]
```diff
@@ -87,8 +87,9 @@ abstract contract BaseHandler is Test {
     bool                     public firstTake;        // if take is called on auction first time
     mapping(address => bool) public alreadyTaken;     // mapping borrower address to true if auction taken atleast once
 
-    string  internal path = "logfile.txt";
-    uint256 internal logFileVerbosity;
+    string  internal path = "logFile.txt";
+    bool    internal logToFile;
+    uint256 internal logVerbosity;
```
Comment (ith-harvey) [inline tests/forge/invariants/base/handlers/unbounded/BaseHandler.sol:92]: initialized this globally to get around stack too deep or else would have kept it local.

File: tests/README.md (hunk 2 above)
Comment (ith-harvey) [reply tests/README.md]: Logging pools is no longer required, fixed here -> https://github.com/ajna-finance/contracts/pull/925/commits/54eadf03a560a53b7afc775979309885d101b76b

**Review by grandizzy — APPROVED** [review]

**Review by prat-gpt — APPROVED** [review]
LGTM

**Pull request merged**